packages/typescript/test/ts_project/allow_js/BUILD.bazel

```diff
@@ -0,0 +1,36 @@
+load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_test")
+load("//packages/typescript:index.bzl", "ts_project")
+
+# Ensure that a.js produces outDir/a.js and outDir/a.d.ts
+SRCS = [
+    "a.js",
+]
+
+ts_project(
+    name = "tsconfig",
+    srcs = SRCS,
+    allow_js = True,
+    declaration = True,
+    declaration_map = True,
+    out_dir = "out",
+    source_map = True,
+)
+
+filegroup(
+    name = "types",
+    srcs = [":tsconfig"],
+    output_group = "types",
+)
+
+nodejs_test(
+    name = "test",
+    data = [
+        ":tsconfig",
+        ":types",
+    ],
+    entry_point = "verify.js",
+    templated_args = [
+        "$(locations :types)",
+        "$(locations :tsconfig)",
+    ],
+)
```

Comment:
Should we use generated_file_test here?

Reply:
I agree it would be better, but not enough to nitpick and make more work for casual contributors